### PR TITLE
[Export] Add extra args for exporter CLI

### PR DESCRIPTION
**Pull request by JingyaHuang**
* Add `compiler_workdir` to store neff files during the compilation
* Add `optlevel` ("O1" / "O2" / "O3")

### CODE REVIEW

File: tests/cli/test_export_cli.py [hunk 1]
```diff
@@ -152,6 +152,57 @@ def test_dynamic_batching(self):
...
+                    "1",
+                    "--task",
+                    "text-classification",
+                    "--compiler_workdir",
+                    f"{tempdir}/neff",
+                    tempdir,
+                ],
+                shell=False,
+                check=True,
+            )
```
Comment (dacorvo) [inline tests/cli/test_export_cli.py:204]: Maybe here you could check that there is a NEFF file in the tempdir/neff directory.

File: tests/inference/test_modeling.py [hunk 2]
```diff
@@ -123,6 +123,15 @@ def test_trust_remote_code(self):
...
         self.assertIsInstance(model.config, PretrainedConfig)
 
+    @requires_neuronx
+    def test_save_compiler_intermediary_files(self):
+        with tempfile.TemporaryDirectory() as tempdir:
+            save_path = f"{tempdir}/neff"
+            _ = NeuronModelForSequenceClassification.from_pretrained(
+                self.MODEL_ID, export=True, compiler_workdir=save_path, **self.STATIC_INPUTS_SHAPES
+            )
+            self.assertTrue(os.path.isdir(save_path))
```
Comment (dacorvo) [inline tests/inference/test_modeling.py:133]: Maybe also check there is at least one neff in it ?

**Review by dacorvo — APPROVED** [review]
Thanks for the pull-request ! LGTM, but maybe you can improve a little the tests.

File: tests/cli/test_export_cli.py (hunk 1 above)
Comment (JingyaHuang) [reply tests/cli/test_export_cli.py:204]: sounds good, I will add a check for that!

File: tests/inference/test_modeling.py (hunk 2 above)
Comment (JingyaHuang) [reply tests/inference/test_modeling.py:133]: same.

**Pull request merged**